operator/cmd/mesh/manifest-validate.go

```diff
@@ -0,0 +1,345 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mesh
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+
+	"istio.io/istio/operator/pkg/util"
+	"istio.io/istio/operator/pkg/validate"
+	"istio.io/istio/pkg/config/validation"
+)
+
+const (
+	traceSamplingMin float64 = 0.0
+	traceSamplingMax float64 = 100.0
+)
+
+var (
+	// Boolean values for --set flags
+	boolValues = []bool{true, false}
+
+	// Ref: https://kubernetes.io/docs/concepts/configuration/overview/#container-images
+	imagePullPolicy = []string{"Always", "IfNotPresent", "Never"}
```

Comment:
We especially don't want to be repeating validations that are already done inside k8s - could we reuse the generated code there to unmarshal the field and pass through the error?
Second best option would be to express this as enum in the proto. We definitely don't want to be copying valid values here as it will create a maintenance problem.

Reply:
I agree a single source of truth avoids maintenance problems. I will see if we can use generated code or enum.